File: backend/src/domain/repositories/ICommunityRepository.ts
import { Community } from '../entities/Community';

export interface ICommunityRepository {
  findById(id: string): Promise<Community | null>;
  findByName(name: string): Promise<Community | null>;
  findNearby(lat: number, lng: number, radiusKm: number): Promise<Community[]>;
  findByCountry(country: string): Promise<Community[]>;
  findAll(): Promise<Community[]>;
  save(community: Community): Promise<Community>;
  update(community: Community): Promise<Community>;
  delete(id: string): Promise<void>;
}

File: backend-dotnet/src/Hestia.Domain/Repositories/ICommunityRepository.cs
using Hestia.Domain.Entities;

namespace Hestia.Domain.Repositories;

public interface ICommunityRepository
{
    Task<Community?> FindByIdAsync(Guid id, CancellationToken ct = default);
    Task<Community?> FindByNameAsync(string name, CancellationToken ct = default);
    Task<IReadOnlyList<Community>> FindNearbyAsync(double lat, double lng, double radiusKm, CancellationToken ct = default);
    Task<IReadOnlyList<Community>> FindByCountryAsync(string country, CancellationToken ct = default);
    Task<IReadOnlyList<Community>> FindAllAsync(CancellationToken ct = default);
    Task<Community> SaveAsync(Community community, CancellationToken ct = default);
    Task<Community> UpdateAsync(Community community, CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}
